### PR TITLE
Honor the NINJA_STATUS environment

### PR DESCRIPTION
**Pull request by vlm**
The `NINJA_STATUS` specifies the prefix for the status line:

https://ninja-build.org/manual.html#_environment_variables

rdar://55968598

### CODE REVIEW

**Comment by vlm** [comment]
@swift-ci please smoke test

File: lib/Commands/NinjaBuildCommand.cpp [hunk 1]
```diff
@@ -718,6 +728,191 @@ class BuildContext : public ExecutionQueueDelegate {
...
+    // Compute the number of max commands to show, subtracting out all the
+    // commands that we avoided running.
+    int possibleMaxCommands = totalPossibleMaxCommands -
+      (numCommandsUpToDate + numCommandsUpdated);
+
+    return possibleMaxCommands;
+  }
+
+  /// Create a resulting byte from the octal components.
+  static bool fromOctalParts(uint8_t& result, char o1, char o2, char o3) {
```
Comment (ddunbar) [inline lib/Commands/NinjaBuildCommand.cpp]: We already have `StringRef::getAsInteger`

File: lib/Commands/NinjaBuildCommand.cpp [hunk 2]
```diff
@@ -718,6 +728,191 @@ class BuildContext : public ExecutionQueueDelegate {
...
+            auto now = std::chrono::steady_clock::now();
+            auto elapsed = now - buildStartTime;
+            auto elapsedMs = std::chrono::duration_cast<std::chrono::milliseconds>(elapsed).count();
+            char buf[64];
+            snprintf(buf, sizeof(buf), "%.3f", (double)elapsedMs / 1000.0);
+            s.append(buf);
+          }
+          break;
+        case 'f':
+          // Numer completed.
```
Comment (ddunbar) [inline lib/Commands/NinjaBuildCommand.cpp]: Sic, number

File: lib/Commands/NinjaBuildCommand.cpp (hunk 1 above)
Comment (vlm) [reply lib/Commands/NinjaBuildCommand.cpp]: Great, thanks!

**Comment by vlm** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**